dotenv-linter/src/schema.rs

```diff
@@ -0,0 +1,310 @@
+use anyhow::Result;
+use serde::Deserialize;
+use std::{fs::File, io::BufReader, path::Path};
+#[derive(Deserialize, Default)]
+#[serde(default)]
+pub struct DotEnvSchema {
+    pub version: String,
+    pub allow_other_keys: bool,
+    pub entries: Vec<SchemaEntry>,
```

Comment:
Yeah, looks like can be represented as a `HasMap` (name -> type & its rules)? In general it also would prevent any possible duplicate definitions